test/LearningTestTool/py/_kht_utils.py

```diff
@@ -27,37 +27,70 @@
 
 def check_test_dir(checked_dir):
     """Test si un chemin est celui d'un repertoire de test"""
-    home_dir = parent_dir_name(checked_dir, 3)
-    tool_dir = parent_dir_name(checked_dir, 2)
-    if home_dir != kht.LEARNING_TEST or tool_dir not in kht.TOOL_DIR_NAMES.values():
+    checked_home_dir_path = parent_dir_path(checked_dir, 3)
+    checked_tool_dir_name = parent_dir_name(checked_dir, 2)
+    if (
+        not check_home_dir(checked_home_dir_path)
+        or checked_tool_dir_name not in kht.TOOL_DIR_NAMES.values()
+    ):
         fatal_error(checked_dir + " should be a test directory of " + kht.LEARNING_TEST)
+    return True
 
 
 def check_suite_dir(checked_dir):
     """Test si un chemin est celui d'un repertoire de suite"""
-    home_dir = parent_dir_name(checked_dir, 2)
-    tool_dir = parent_dir_name(checked_dir, 1)
-    if home_dir != kht.LEARNING_TEST or tool_dir not in kht.TOOL_DIR_NAMES.values():
+    checked_home_dir_path = parent_dir_path(checked_dir, 3)
+    checked_tool_dir_name = parent_dir_name(checked_dir, 1)
+    if (
+        not check_home_dir(checked_home_dir_path)
+        or checked_tool_dir_name not in kht.TOOL_DIR_NAMES.values()
+    ):
         fatal_error(
             checked_dir + " should be a suite directory of " + kht.LEARNING_TEST
         )
+    return True
 
 
 def check_tool_dir(checked_dir):
     """Test si un chemin est celui d'un repertoire d'outil"""
-    home_dir = parent_dir_name(checked_dir, 1)
-    tool_dir = parent_dir_name(checked_dir, 0)
-    if home_dir != kht.LEARNING_TEST or tool_dir not in kht.TOOL_DIR_NAMES.values():
+    checked_home_dir_path = parent_dir_path(checked_dir, 3)
+    checked_tool_dir_name = parent_dir_name(checked_dir, 0)
+    if (
+        not check_home_dir(checked_home_dir_path)
+        or checked_tool_dir_name not in kht.TOOL_DIR_NAMES.values()
+    ):
         fatal_error(checked_dir + " should be a tool directory of " + kht.LEARNING_TEST)
+    return True
 
 
-def check_home_dir(checked_dir):
+def check_home_dir(checked_dir, fatal_error_if_false=True):
     """Test si un chemin est celui du repertoire LearningTest"""
-    home_dir = parent_dir_name(checked_dir, 0)
-    if home_dir != kht.LEARNING_TEST:
+    checked_home_dir_path = parent_dir_path(checked_dir, 0)
+    # On n'impose pas que le repertoire racine ait un nom predefini
+    fixed_home_dir_name = False
+    if fixed_home_dir_name:
+        checked_home_dir_name = os.path.basename(checked_home_dir_path)
+        if checked_home_dir_name != kht.LEARNING_TEST:
+            if fatal_error_if_false:
+                fatal_error(
+                    checked_dir + " should be the '" + kht.LEARNING_TEST + "' directory"
+                )
+            return False

```

Comment:
Is this still necessary? The `if` is never executed.

Reply:
ok: ce code désormais obsolète est supprimé